первые 3 задачи

durf666 opened the pull request:
Сделал первые 3 задачи к 7 занятию. spinner, pipeline и buf_chan соответственно

durf666 committed:
первые 3 задачи

shmel1k commented on time_server/main.go:
```diff
@@ -0,0 +1,66 @@
+		bytes, _ := bufReader.ReadBytes('\n')
+		fmt.Println(bytes)
+		b := []byte{101, 120, 105, 116, 13, 10}
+		if Equal(bytes, b) {
```
есть bytes.Equals(или Equal, не помню)

shmel1k commented on time_server/main.go:
```diff
@@ -0,0 +1,66 @@
+	for {
+
+		bufReader := bufio.NewReader(c)
+		bytes, _ := bufReader.ReadBytes('\n')
```
Если клиент отключится сам, то есть большая вероятность, что эта горутина утечёт

durf666 committed:
сделал ДЗ 7

Сделал задачки 4 и 5 (buf_chan и race) и поправил третью (time_server) теперь по команде вырубает сам сервер. Вопрос по третьей. Вы писали, что горутина  может  утечь если клиент закроет соединение, как это пофиксить?

shmel1k commented on pipeline/main.go:
```diff
@@ -0,0 +1,38 @@
+		close(squares)
+	}()
+
+	for {
+		x, ok := <-squares
+		if !ok {
```
for v := range squares {}

shmel1k commented on race/main.go:
```diff
@@ -0,0 +1,73 @@
+					fmt.Println("car is ready.")
+					wg.Done()
+				} else {
+					fmt.Println("Channel closed!")
```
А если канал закрылся, то этот цикл будет крутиться бесконечно?
А если канал открыт, то будет ли этот цикл крутиться бесконечно, потребляя 100% CPU?

shmel1k commented on race/main.go:
```diff
@@ -0,0 +1,73 @@
+	go func() {
+		for {
+			select {
+			case x, ok := <-raceT:
```
Аналогично вопросу выше

durf666 committed:
допилил гонку
